yt/frontends/gizmo/data_structures.py

```diff
@@ -58,3 +61,12 @@ def _is_valid(self, *args, **kwargs):
             valid = False
             pass
         return valid
+
+    def _set_code_unit_attributes(self):
+        super(GizmoDataset, self)._set_code_unit_attributes()
+        munit = np.sqrt(self.mass_unit /
+                        (self.time_unit**2 * self.length_unit)).to("gauss")
+        if self.cosmological_simulation:
+            self.magnetic_unit = self.quan(munit.value, "%s/a**2" % munit.units)
```

Comment:
@tkc004 might want to have a look if this definition of code magnetic unit is correct.

Reply:
Magnetic fields in the output should be in unit of ``gauss'' (physical). We do not need to convert it from mass_unit, time_unit, etc. Even in cosmological runs, the magnetic field is in ``gauss'' (physical) and no a is needed.

Reply:
scripts/gizmo_documentation.html:MAGNETIC_FIELD_code = UnitMagneticField_in_gauss = (params.txt default = 1 Gauss)  
scripts/gizmo_documentation.html:MAGNETIC_FIELD_physical = MAGNETIC_FIELD_code  
scripts/gizmo_documentation.html:MAGNETIC_FIELD_physical = MAGNETIC_FIELD_code (note that *in* the code, co-moving units are used B_code=a_scale^2 B_phys, but in outputs these are converted to physical)

Reply:
@tkc004 so will this always be true or is the unit user-configurable? And if it is configurable can we read it from the output?

Reply:
I have some thoughts about this. All the code units in GIZMO (and other Gadget derivatives as well I believe) are [configurable in the parameter file](http://www.tapir.caltech.edu/~phopkins/Site/GIZMO_files/gizmo_documentation.html#params-generic-units) using constants like `UnitMagneticField_in_gauss`. But this information is not available in the output snapshots. Up until now, the code units in these frontends are set according to the defaults from the official documentation. That is fine because the code units could also be set manually if needed. The most important thing might be making sure the users are aware of this. Personally, I think it could be convenient to have an interface to set up code unit systems from the same set of constants the code requires like `ds.set_unit_system(UnitLength_in_cm, UnitMass_in_g, ...)`. @ngoldbaum's unyt PR might also be related to this discussion.